kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaTopicManagerSharedState.java

```diff
@@ -84,44 +94,29 @@ private void cancelCursorExpireTask() {
         }
     }
 
-    private void removePersistentTopicAndReferenceProducer(final String topicName) {
-        // 1. Remove PersistentTopic and Producer from caches, these calls are thread safe
-        final CompletableFuture<Optional<PersistentTopic>> topicFuture = topics.remove(topicName);
-        final Producer producer = references.remove(topicName);
-
-        if (topicFuture == null) {
-            KopBrokerLookupManager.removeTopicManagerCache(topicName);
-            return;
-        }
+    public Optional<Producer> registerProducer(String topic, KafkaRequestHandler requestHandler,
+                                               Supplier<Producer> supplier) {
```

Comment:
Why use Supplier<Producer> here?

Reply:
because you want to create the Producer only if needed, and not everytime

Reply:
I see.